### DISCUSSION TITLE
Best way to increment/decrement a field?

### DISCUSSION THREAD

**Discussion by jamesonev** [prisma/prisma#3176] [Q&A] [answered]
What's the best practice to decrement a field in one of my records? I took a look at [#266 (from prisma1)](https://github.com/prisma/prisma1/issues/266) but couldn't find it documented anywhere. I'm just trying to atomically decrement a field that I know exists, ideally in a single query.

**Comment by ryands17** [comment, marked as answer]
Hey @jamesonev 👋 
Unfortunately this is not possible directly and there is an open issue for that as well [here](https://github.com/prisma/prisma-client-js/issues/775). Do add a 👍 on this so that we can measure the priority.

As a workaround, you would need to use `prisma.executeRaw` as follows. Here I am incrementing the Post's views.

```ts
await prisma.executeRaw(`update "Post" set views = views + 1 where id = 1`)
```

**Reply by moishinetzer** [reply]
This is now natively supported by ```increment: x``` since 2.6.0

Check out the release notes: https://github.com/prisma/prisma/releases/tag/2.6.0

**Reply by skillradius360** [reply]
various codebases still have this deprecated value still..thanks for the help guis

**Comment by LeandreArnaud** [comment]
Hi @jamesonev ,
It seems that Prisma heard @ryands17 👍 :
```ts
prisma.table.update({
      where: { id: 1234 },
      data: {views: {increment: 1}}
    });
```

**Reply by bluuewhale** [reply]
awesome

**Reply by mualle** [reply]
great!

**Reply by camilo86** [reply]
This is amazing

**Reply by opaque-maniac** [reply]
Thanks

**Comment by kingfishx23** [comment]
I usually use:
```
Increment:
prisma.table.update({
  where: { id: 1234 },
  data: {views: {increment: 1}}
});

Decrement:
prisma.table.update({
  where: { id: 1234 },
  data: {views: {increment: -1}}
});
```

**Reply by ujjwal-kr** [reply]
there is a `decrement` by now

**Reply by janpio** [reply]
Indeed: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#operators

**Comment by leppaott** [comment]
How to set it to zero on start?

**Reply by AbdelrahmanAbounida** [reply]
when initiate the  model u can set `@default(0) `

**Reply by leppaott** [reply]
@AbdelrahmanAbounida that's what I ended up with but you can't default it while incrementing. `OR: {increment: 1} {set: 1}`

**Reply by charanjit-singh-1** [reply]
Due to some reasonn

<img width="381" alt="image" src="https://github.com/user-attachments/assets/51832ccd-5e54-409b-b5c3-d3bd966540be">


not working

i changed BigInt to Int and after that it stopped working.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
